src/api/app/components/watched_items_list_component.html.haml

```diff
@@ -3,29 +3,42 @@
   - @items.each do |item|
     - case @class_name
       - when 'Package'
+        - name = "#{item.project}/#{item}"
         .d-flex.flex-row.flex-wrap.align-items-baseline
           = link_to(package_show_path(item.project, item), class: 'text-word-break-all') do
             %i.fas.fa-archive.mr-1
-            #{item.project}/#{item}
-          = link_to(project_package_toggle_watched_item_path(project_name: item.project.name, package_name: item.name),
-                    method: :put,
-                    class: 'text-light ml-auto') do
-            %i.fa.fa-times-circle
+            = name
+          = link_to('#', class: 'text-light ml-auto',
+                    data: { toggle: 'modal',
+                            target: '#delete-item-from-watchlist-modal',
+                            confirmation_text: "Please confirm you want to remove '#{name}' from the watchlist.",
+                            action: project_package_toggle_watched_item_path(project_name: item.project.name, package_name: item.name) }) do
+            %i.fa.fa-times-circle{ title: 'Remove package from watchlist' }
       - when 'Project'
+        - name = item.name
         .d-flex.flex-row.flex-wrap.align-items-baseline
           = link_to(project_show_path(item), class: 'text-word-break-all') do
             %i.fas.fa-cubes.mr-1
-            #{item}
-          = link_to(project_toggle_watched_item_path(project_name: item.name), method: :put, class: 'text-light ml-auto') do
-            %i.fa.fa-times-circle
+            = name
+          = link_to('#', class: 'text-light ml-auto',
+                    data: { toggle: 'modal',
+                            target: '#delete-item-from-watchlist-modal',
+                            confirmation_text: "Please confirm you want to remove '#{name}' from the watchlist.",
+                            action: project_toggle_watched_item_path(project_name: name) }) do
```

Comment:
@saraycp maybe you can also set the modal title here, would be a bit nicer to mention the type as well. So instead of `'Do you really want to remove this item?'` maybe something like `'Do you really want to remove this #{type}?'

Reply:
Good suggestion. Done